paddle/strings/stringpiece.h

```diff
@@ -0,0 +1,113 @@
+/*
+  Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#pragma once
+
+#include <assert.h>
+#include <string.h>
+
+#include <stdexcept>
+#include <string>
+
+namespace paddle {
+
+// StringPiece points into a std::string object but doesn't own the
+// string.  It is for efficient access to strings.  Like Go's string
+// type.  StringPiece is not thread-safe.
```

Comment:
Since StringPiece is immutable, it would be thread-safe as long as the underlaying string is not mutated during the lifetime of StringPiece.

Maybe we could change "StringPiece is not thread-safe." to "StringPiece is thread-safe as long as the underlying string is not mutated."?